alchemy/src/cloudflare/bucket.ts

```diff
@@ -302,13 +302,45 @@ export function withJurisdiction(
   return result;
 }
 
+/**
+ * Get a bucket
+ */
+export async function getBucket(
+  api: CloudflareApi,
+  bucketName: string,
+  props: BucketProps = {}
+): Promise<CloudflareBucketResponse> {
+  const headers = withJurisdiction({}, props);
+  const getResponse = await api.get(
+    `/accounts/${api.accountId}/r2/buckets/${bucketName}`,
+    { headers }
+  );
+
+  if (!getResponse.ok) {
+    return await handleApiError(getResponse, "get", "R2 bucket", bucketName);
+  }
+
+  if (getResponse.status === 200) {
+    return (await getResponse.json()) as CloudflareBucketResponse;
+  }
+
+  const errorData: any = await getResponse.json().catch(() => ({
+    errors: [{ message: getResponse.statusText }],
+  }));
+
+  throw new CloudflareApiError(
+    `Error getting R2 bucket '${bucketName}': ${errorData.errors?.[0]?.message || getResponse.statusText}`,
+    getResponse
+  );
+}
+
 /**
  * Create a new bucket
  */
 export async function createBucket(
   api: CloudflareApi,
   bucketName: string,
-  props: BucketProps
+  props: BucketProps = {}
```

Comment:
Everything's optional in `BucketProps` interface.

I'm happy to remove this if the parent call to `createBucket` should be passing something for props (e.g. `{ name: this.bucketName }`)

Reply:
Nope, this is good. I've been making the same changes, defaulting name to id so we can create resources with no props.